Documentation/templates/strings/objc.md

```diff
@@ -0,0 +1,53 @@
+## Template Information
+
+| Name      | Description       |
+| --------- | ----------------- |
+| File names | strings/objc-string-h.stencil & strings/objc-string-m.stencil |
+| Configuration example | <pre>strings:<br />  inputs: path/to/Localizable.strings<br />  outputs:<br />    - templateName: objc-string-h<br />      output: Localizable.h</pre><br />    - templateName: objc-string-m<br />      output: Localizable.m</pre> |
+| Language | Objective-C |
+| Author | Eric Slosser |
+
+## When to use it
+
+- When you need to generate *Objective-C* code.
+
+## Customization
+
+You can customize some elements of this template by overriding the following parameters when invoking `swiftgen`. See the [dedicated documentation](../../ConfigFile.md).
+
+| Parameter Name | Default Value | Description |
+| -------------- | ------------- | ----------- |
+| `noComments`   | N/A           | Setting this parameter will disable the comments describing the translation of a key. |
+| `headerName`   | "Localizable" | A generated .m file `#includes "{headerName}.h"` |
+
+## Generated Code
+
+**Extract:**
+
+```swift
+@interface Localizable : NSObject
+// alert__message --> "Some alert body there"
++ (NSString*)alertMessage;
+// alert__title --> "Title of the alert"
++ (NSString*)alertTitle;
+// ObjectOwnership --> "These are %3$@'s %1$d %2$@."
++ (NSString*)objectOwnership:(NSInteger)p1 and:(NSString*)p2 and:(NSString*)p3;
```

Comment:
I know that this is some artefact of ObjC, but those `and:… and:… and:…` are not great. I forgot that ugly part of ObjC I guess 😓 
An alternative could be to not give any name to those parameters, which is technically allowed in ObjC… even if not common at all (it's only encountered on like 3 or 4 functions in Apple's SDK iirc);
```objc
+ (NSString*)objectOwnership:(NSInteger)p1 :(NSString*)p2 :(NSString*)p3;
```
And it's used like this in code:
```objc
NSString* fruit = @"bananas";
NSString* name = @"Olivier";
label.text = [Localizable objectOwnership:5:fruit:name];
```

Not saying that it's better. In fact it's so uncommon that it might feel uglier than the version with `and:… and:… and:…". But just thought I'd mention that this could be another possibility / another route we could choose too. Matter of taste :wink:

Reply:
Forgot that was even possible, read it in some obscure article somewhere I think.

Would the compiler even know how to autocomplete that correctly? The `and...and` version'll work fine. This, not so sure?

Reply:
I think it will. It's used in the SDK at least by the `+[CAMediaTimingFunction functionWithControlPoints::::]` method, you could experiment with that if you want 😉

Reply:
Note: if we still go with named parameters I think `with:` would generally be a better fit than `and:`? (But I have to admit I don't remember the coding guidelines for ObjC that well 😅)

Reply:
I think `with:...` is better than `and:...`.   I think they both are references to existing (non-auto-generated) signatures that look like `-[doSomethingWith:and:and:]`. 

I originally had `p1:p2:p2:`, but that's not great either, and I was happy to switch away from it.

Reply:
Is there a consensus?

Reply:
Looking at the media timing signature, it'd have to be `...WithValues: : : :`. (could be "values", "arguments", ...)

Reply:
that works for me.

Reply:
I'll do the test for count==1, so we'll have `...WithValue:` and `...WithValues:`.  It makes for an uglier stencil, but prettier generated code.